kerastuner/applications/augment.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+from kerastuner.engine import hypermodel
+
+from tensorflow.keras import layers
+from tensorflow.keras.layers.experimental import preprocessing
+
+import random
+
+# dict of functions that create layers for transforms.
+# Each function takes a factor (0 to 1) for the strength
+# of the transform.
+TRANSFORMS = {
+'translate_x': lambda x: preprocessing.RandomTranslation(x, 0),
+'translate_y': lambda y: preprocessing.RandomTranslation(0, y),
+'rotate': preprocessing.RandomRotation,
+'contrast': preprocessing.RandomContrast,
+}
+
+class HyperAugment(hypermodel.HyperModel):
+    """ Builds HyperModel for image augmentation.
+
+    # Arguments:
+        input_shape: Optional shape tuple, e.g. `(256, 256, 3)`.
+        input_tensor: Optional Keras tensor (i.e. output of
+            `layers.Input()`) to use as image input for the model.
+        rotate: Whether to include rotation in augmentation.
+        translate_x: Whether to include horizontal translation in augmentation.
+        translate_y: Whether to include vertical translation in augmentation.
+        contrast: Whether to include random contrast in augmentation.
+        **kwargs: Additional keyword arguments that apply to all
+            HyperModels. See `kerastuner.HyperModel`.
+    """
+    def __init__(self,
+        input_shape=None,
+        input_tensor=None,
+        rotate=True,
+        translate_x=True,
+        translate_y=True,
+        contrast=True):
+        self.transforms = []
+        if rotate:
+            self.transforms.append('rotate')
+        if translate_x:
+            self.transforms.append('translate_x')
+        if translate_y:
+            self.transforms.append('translate_y')
+        if contrast:
+            self.transforms.append('contrast')
+
+        self.input_shape = input_shape
```

Comment:
Do u need to check whether only one of the input_shape nad input_tensor is provided?

Reply:
`HyperFixedAugment` only uses sequential model so `input_shape` and `input_tensor` can be both missing. So the check is done only for `HyperRandAugment`.